python-backend/app/api/documents/expected/resources/mine_documents.py

```diff
@@ -47,6 +48,8 @@ def post(self, mine_guid):
         data = self.parser.parse_args()
         doc_list = data['documents']
         mine_new_docs = []
+        doc_status_list = ExpectedDocumentStatus.find_all_document_status()
+        not_received = next(filter(lambda x: x.description == 'Not Received', doc_status_list))
```

Comment:
If "Not Received" is a special kind of status, perhaps it deserves its own method on the model. I'm not sure this is appropriate logic to have in the resource

Reply:
There is currently a ticket in the backlog to refactor all of this as the ExpectedDocumtneStatus is still using a guid instead of a 3 character code. That refactor was decided to be out of scope for this ticket, and that should handle this issue.

Reply:
I don't see how this is related to using a guid vs a 3 character code? This check is being done against the description, and the logic is happening in the resource. My request was just that it be moved to the model

Reply:
I originally misread the question, I'm sorry for the confusion on my part. I understand what your asking and I have added a new function on the model to find_by_description as doing that at the SQL level will be far more efficient than filtering the entire returned list.